src/mesh/wifi/WiFiAPClient.cpp

```diff
@@ -153,24 +157,38 @@ static int32_t reconnectWiFi()
         isReconnecting = true;
 
         // Make sure we clear old connection credentials
-#ifdef ARCH_ESP32
+    #ifdef ARCH_ESP32
         WiFi.disconnect(false, true);
-#elif defined(ARCH_RP2040)
+    #elif defined(ARCH_RP2040)
         WiFi.disconnect(false);
-#endif
+    #endif
         LOG_INFO("Reconnecting to WiFi access point %s", wifiName);
 
-        delay(5000);
+        // Start the non-blocking wait for 5 seconds
+        wifiReconnectStartMillis = millis();
+        wifiReconnectPending = true;
+        // Do not attempt to connect yet, wait for the next invocation
+        return 100; // Schedule next check soon
+    }
 
-        if (!WiFi.isConnected()) {
-#ifdef CONFIG_IDF_TARGET_ESP32C3
-            WiFi.mode(WIFI_MODE_NULL);
-            WiFi.useStaticBuffers(true);
-            WiFi.mode(WIFI_STA);
-#endif
-            WiFi.begin(wifiName, wifiPsw);
+    // Check if we are ready to proceed with the WiFi connection after the 5s wait
+    if (wifiReconnectPending) {
+        if (millis() - wifiReconnectStartMillis >= 5000) {
+            if (!WiFi.isConnected()) {
+            #ifdef CONFIG_IDF_TARGET_ESP32C3
+                WiFi.mode(WIFI_MODE_NULL);
+                WiFi.useStaticBuffers(true);
+                WiFi.mode(WIFI_STA);
+            #endif
+                WiFi.begin(wifiName, wifiPsw);
+            }
+            isReconnecting = false;
+            wifiReconnectPending = false;
+            return 300000; // Resume normal interval after reconnect
```

Comment:
I think it's better not to return here, as it should e.g. execute `onNetworkConnected()` after this, and not only after 5 min.